crates/control/src/team_ball_receiver.rs

```diff
@@ -0,0 +1,155 @@
+use std::time::{Duration, SystemTime};
+
+use color_eyre::eyre::Result;
+use context_attribute::context;
+use coordinate_systems::{Field, Ground};
+use linear_algebra::{Isometry2, Point2, Vector2};
+use serde::{Deserialize, Serialize};
+
+use framework::{AdditionalOutput, MainOutput, PerceptionInput};
+use spl_network_messages::{GamePhase, HulkMessage, SubState};
+use types::{
+    ball_position::BallPosition, cycle_time::CycleTime,
+    filtered_game_controller_state::FilteredGameControllerState, messages::IncomingMessage,
+    players::Players,
+};
+
+#[derive(Deserialize, Serialize)]
+pub struct TeamBallReceiver {
+    received_balls: Players<Option<BallPosition<Field>>>,
+}
+
+#[context]
+pub struct CreationContext {}
+
+#[context]
+pub struct CycleContext {
+    cycle_time: Input<CycleTime, "cycle_time">,
+    filtered_game_controller_state:
+        Input<Option<FilteredGameControllerState>, "filtered_game_controller_state?">,
+    ground_to_field: Input<Option<Isometry2<Ground, Field>>, "ground_to_field?">,
+    network_message: PerceptionInput<Option<IncomingMessage>, "SplNetwork", "filtered_message?">,
+
+    striker_trusts_team_ball: Parameter<Duration, "spl_network.striker_trusts_team_ball">,
+
+    team_balls: AdditionalOutput<Players<Option<BallPosition<Field>>>, "team_balls">,
+}
+
+#[context]
+#[derive(Default)]
+pub struct MainOutputs {
+    pub team_ball: MainOutput<Option<BallPosition<Field>>>,
+    pub network_robot_obstacles: MainOutput<Vec<Point2<Ground>>>,
```

Comment:
I'm fine with the current solution, but this may also be moved to a completely independent separate Node...
What do you think?

Reply:
Yes, was planning on doing that; will do.